### PR TITLE
Add option for WAL compression algorithm

### PR DESCRIPTION
**Pull request by sidroyc**
Summary: Add an option to set the WAL compression algorithm - wal_compression. 

TODO: WAL compression is not implemented and will only support zstd initially. Will be added in subsequent diffs.

### CODE REVIEW

File: db/db_impl/db_impl.h [hunk 1]
```diff
@@ -2287,6 +2287,7 @@ class DBImpl : public DB {
   // In 2PC these are the writes at Prepare phase.
   const bool two_write_queues_;
   const bool manual_wal_flush_;
+  const CompressionType wal_compression_;
```
Comment (pdillinger) [inline db/db_impl/db_impl.h:2290]: I don't think this is needed because it's part of `immutable_db_options_`

File: include/rocksdb/options.h [hunk 2]
```diff
@@ -1228,6 +1228,9 @@ struct DBOptions {
   // file.
   bool manual_wal_flush = false;
 
+  // If enabled WAL records will be compressed before they are written.
```
Comment (pdillinger) [inline include/rocksdb/options.h:1231]: This needs to be marked EXPERIMENTAL or WORK IN PROGRESS

**Review by pdillinger — CHANGES_REQUESTED** [review]
Looking good, thanks

File: include/rocksdb/options.h [hunk 3]
```diff
@@ -1228,6 +1228,9 @@ struct DBOptions {
...
+  CompressionType wal_compression = kNoCompression;
```
Comment (anand1976) [inline include/rocksdb/options.h:1232]: Mention that only zstd is supported.

File: db/db_impl/db_impl_open.cc [hunk 4]
```diff
@@ -194,6 +194,16 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src,
   }
 #endif  // !ROCKSDB_LITE
 
+  // Supported wal compression types
```
Comment (anand1976) [inline db/db_impl/db_impl_open.cc:197]: We can just check ```result.wal_compression``` rather than going the roundabout way.

**Review by anand1976 — COMMENTED** [review]
Thanks for the PR! Looks good overall.

File: include/rocksdb/options.h (hunk 2 above)
Comment (sidroyc) [reply include/rocksdb/options.h:1232]: Addressed.

File: include/rocksdb/options.h (hunk 3 above)
Comment (sidroyc) [reply include/rocksdb/options.h:1234]: Addressed.

File: db/db_impl/db_impl_open.cc (hunk 4 above)
Comment (sidroyc) [reply db/db_impl/db_impl_open.cc:197]: Addressed.

File: db/db_impl/db_impl.h (hunk 1 above)
Comment (sidroyc) [reply db/db_impl/db_impl.h]: Addressed.

**Comment by sidroyc** [comment]
@anand1976 @pdillinger Thanks for the review! Addressed all your comments.

**Review by anand1976 — APPROVED** [review]
LGTM

**Pull request closed without merging**